### PR TITLE
add documentation for how awx uses/interacts with ansible

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# awx
+awx provides a web interface and distributed task engine for scheduling and
+running Ansible playbooks.  As such, it relies heavily on the interfaces
+provided by Ansible.  This document provides a birds-eye view of the notable
+touching points between awx and Ansible.
+
```
Comment (wenottingham) [inline docs/overview.md]: touchpoints?

File: docs/overview.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  arguments (extra_vars, verbosity, forks, limit, etc...)
+
+### Credentials
+awx stores sensitive credential data which can be attached to `ansible-playbook`
+processes that it runs.  This data can be oriented towards SSH connection
+authentication (usernames, passwords, SSH keys and passphrases),
+ansible-specific prompts (such as Vault passwords), or environmental
+authentication values which various Ansible modules depend on (such as setting
+`AWS_ACCESS_KEY_ID`).
+
```
Comment (wenottingham) [inline docs/overview.md]: extra_vars as well (for custom credentials)

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+mercurial, or subversion repositories).  This import is accomplished via an
+ansible playbook included with awx (which makes use of the various source
+control management modules in Ansible).
+
+### Inventories
+awx manages Inventories Groups, and Hosts, and provides a RESTful interface
+that maps to static and dynamic Ansible inventories.  Inventory data can
+be entered into awx manually, but many users perform Inventory Syncs to import
+inventory data from a variety of external sources.
+
```
Comment (chrismeyersfsu) [inline docs/overview.md]: * control inventory cache enable/disable
* enabled variable
* unique instance id
* control certs/ssl checks

File: docs/overview.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+### Capturing Event Data
+awx applies an Ansible callback plugin to all `ansible-playbook` and `ansible`
+processes it spawns.  This allows Ansible events to be captured and persisted
+into the awx database; this process is what drives the "streaming" web UI
+you'll see if you launch a job from the awx web interface and watch its results
+appears on the screen.  awx relies on stability in this plugin interface, the
+heirarchy of emitted events based on strategy, and _especially_ the structure
+of event data to work across Ansible releases.
+
```
Comment (chrismeyersfsu) [inline docs/overview.md]: Event Hierarchy Linkage
ansible callback events job events - event in tower 5

**Review by matburt — APPROVED** [review]

**Pull request merged**